src/Collector/MetricCollectorInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
```

Comment:
What is the collector for? We have MergingMetricSource and so on, so why we need this interface?

Reply:
For me collector represents higher level lazy abstraction on how and when we obtain the final source. We can remove it, but then every source would be the same rank and the hierarchy should be manually reviewed.
I.e I have added profiler for bundle and there is no real sense in profiling each source in particular, but only top-level sources which are collected (since they all include other metrics which should be collected at the end)
Also not every source is getting merged from low-level sources, with collector it's possible to use it as a metric source factory eliminating intermediate factory services

But question is fully valid, currently we can easily remove the collector layer and everything would  work fine, just with a bit complicated configuration